feat: adds new github-env audit (#192)
woodruffw/zizmor: docs/audits.md

        echo "$HOME/.local/my-bin" >> "$GITHUB_PATH"
    ```

## `github-env`

| Type     | Examples           | Introduced in | Works offline  | Enabled by default |
|----------|--------------------|---------------|----------------|--------------------|
| Workflow  | [github-env.yml]   | v0.6.0        | ✅             | ✅                 |

[github-env.yml]: https://github.com/woodruffw/gha-hazmat/blob/main/.github/workflows/github-env.yml

> Detects dangerous usages of `$GITHUB_ENV` environment variable.

Reviewer: ```suggestion
Detects dangerous usages of the `GITHUB_ENV` environment variable.
```

Author: Done [here](https://github.com/woodruffw/zizmor/pull/192/commits/f222f8655f3f0d4a184982ef551822fbd9c07fea) with all suggestions for docs 🙂

> This audit compounds over `dangerous-triggers`: given the proper circumstances, an attacker
> may achieve code execution by controlling this environment variable.

Reviewer: ```suggestion
When used in workflows with dangerous triggers (such as `pull_request_target` and `workflow_run`), `GITHUB_ENV` can be an arbitrary code execution risk.

In particular, if the attacker is able to set arbitrary variables or variable contents via `GITHUB_ENV`, they made be able to set `LD_PRELOAD` or otherwise induce code execution implicitly within subsequent steps.
```

Other resources:

* [GitHub Actions exploitation: environment manipulation]
* [GHSL-2024-177: Environment Variable injection in an Actions workflow of Litestar]

### Remediation

> In general, you should avoid using `$GITHUB_ENV` when relying on dangerous Workflow triggers,
> especially for use cases like caching state or passing state around. For those, `$GITHUB_OUTPUT`
> may be an alternative.
> 

Reviewer: ```suggestion
In general, you should avoid setting `GITHUB_ENV` within workflows that are attacker-triggered, like `pull_request_target`.

If you need to pass state between steps, consider using `GITHUB_OUTPUT` instead.
```
